### PR TITLE
[docs] Add new pages for release versioning and supported releases

### PR DESCRIPTION
**Pull request by stevebang**
This PR includes the following, based on discussions with @chirag-yb and @rkarthik:
- Release versioning page
- Supported and planned releases page
- Notes added to release notes for 2.3, 2.2.2, and 2.2, with links to these pages.
- Note added to listing of earlier releases.

### CODE REVIEW

File: docs/content/latest/releases/releases-overview.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+| 2.4                                    | 2020/11 (expected) |
+| [2.2](../earlier-releases/v2.2.0)      | 2020/07/15 |
+| [2.1](../earlier-releases/v2.1.0)      | 2020/02/25 |
+| [2.0](../earlier-releases/v2.0.0)      | 2019/09/17 |
+| [1.3](../earlier-releases/v1.3.0)      | 2019/07/19 |
+
+## Latest releases
+
+The latest release series specified below is a work-in-progress and introduces
+new features and changes that may break backward compatibility. Use the latest releases
```
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:41]: @stevebang I wouldn't call "changes that may break backward compatibility" anywhere instead say it's not recommended for any production use cases and it's subjected to changes. This will need changes in multiple places

File: docs/content/latest/releases/versioning.md [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+- `MINOR` — Incremented when new, backwards-compatible functionality is introduced.
+  - `EVEN` — Stable minor release, intended for production deployments.
+  - `ODD` — Latest minor release, intended for development and testing.
+- `PATCH` — bug fixes and revisions that do not break backwards compatibility.
+
+Examples follow in the relevant sections below.
+
+## Stable releases
+
+Stable release series, denoted by `MAJOR.EVEN` numbers, introduce fully tested new features and changes added since the last stable release and may break backwards compatibility. The current stable `2.2` release series will be followed by the next stable `2.4` release series.
```
Comment (chirag-yb) [inline docs/content/latest/releases/versioning.md:31]: @stevebang This is not correct "since the last stable release and may break backward compatibility"

File: docs/content/latest/releases/releases-overview.md (hunk 1 above)
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md]: Updated all instances to remove that phrase.

File: docs/content/latest/releases/versioning.md (hunk 2 above)
Comment (stevebang) [reply docs/content/latest/releases/versioning.md]: Fixed.

**Review by chirag-yb — CHANGES_REQUESTED** [review]
@stevebang Could you address two comments and resend it?

File: docs/content/latest/releases/releases-overview.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+**Note:** Versions `2.1`, `2.0`, and `1.3` that were released before `2.2` are also considered stable
+releases and are supported for existing production deployments.
+
+| Version | Released |
+| :------ | -------- |
+| 2.4                                    | 2020/11 (expected) |
+| [2.2](../earlier-releases/v2.2.0)      | 2020/07/15 |
+| [2.1](../earlier-releases/v2.1.0)      | 2020/02/25 |
+| [2.0](../earlier-releases/v2.0.0)      | 2019/09/17 |
+| [1.3](../earlier-releases/v1.3.0)      | 2019/07/19 |
```
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:36]: @stevebang Need to confirm if 1.3 and 2.0 are still supported ? if so, until when? This is going to be written in our support policy 

*****
  Release Support. 
(a) For the maintenance support period of at least 1 year from the major release date, Yugabyte will provide Updates for such release.

(b) Following the maintenance support period, Yugabyte will provide Updates for at least an additional 180 days subject to the following guidelines:
Updates and Upgrades will not be made to the major release.
Yugabyte will direct Customers to existing Updates and workarounds applicable to the reported case.
Yugabyte may direct Customers to Upgrade to a current release if a workaround does not exist.

(c)  Yugabyte will post publicly on its website a notice of End of Life (EOL) for the affected Software and the timeline for discontinuing Support Services. Yugabyte has no obligation to provide Support Services for Software after EOL.

Current Supported Software Releases are listed on Yugabyte’s website.
*****

**Review by chirag-yb — COMMENTED** [review]
@stevebang Need to confirm 1.3 and 2.0 are still supported

File: docs/content/latest/releases/releases-overview.md (hunk 3 above)
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:36]: Based on the policy statement above, both of those are still supported. 1.3 would fall under the post-maintenance support time period. I'll keep my eye out for a link to the support policy and add that when it is available.

**Review by chirag-yb — APPROVED** [review]
@stevebang could you check my last comment? Otherwise, it's good to go from my end. 
Waiting for Karthik's approval

File: docs/content/latest/releases/releases-overview.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+Included below are tentative release dates for upcoming planned releases, subject to change.
+For updates on what's being worked on, see [Current roadmap](https://github.com/yugabyte/yugabyte-db#current-roadmap)
+in the YugabyteDB GitHub repository README page.
+
+| Version | Planned release |
+| :------ | --------------- |
+| 2.4     | 2020/11         |
+| 2.5     | 2021/01         |
+| 2.6     | 2021/03         |
```
[the remark this answers is not in the document]
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:57]: Check the Google Doc — I think I have it correct.

File: docs/content/latest/releases/releases-overview.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+showAsideToc: true
+---
+
+{{< note title="Important" >}}
+
+As of version `2.2`, Yugabyte is following a new release versioning convention for YugabyteDB releases.  For details, see [Release versioning](../versioning).
+
+{{< /note >}}
+
+## Supported stable releases
```
Comment (rkarthik007) [inline docs/content/latest/releases/releases-overview.md:21]: My preference is to combine this table and the one in section below. Just add a column that outlines the recommended latest and stable release. That way, there is one table. This fragmentation is confusing.

Also, a couple of lines below that table, helping the end user quickly decide which release they should pick would be great.

File: docs/content/latest/releases/releases-overview.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+## Planned releases
+
+Included below are tentative release dates for upcoming planned releases, subject to change.
+For updates on what's being worked on, see [Current roadmap](https://github.com/yugabyte/yugabyte-db#current-roadmap)
+in the YugabyteDB GitHub repository README page.
+
+| Version | Planned release |
+| :------ | --------------- |
+| 2.4     | 2020/11         |
+| 2.5     | 2021/01         |
```
Comment (rkarthik007) [inline docs/content/latest/releases/releases-overview.md:56]: These should be flipped. We will put out the 2.4 release in Jan.

**Review by rkarthik007 — COMMENTED** [review]
Some comments... should we do a live session?

File: docs/content/latest/releases/releases-overview.md (hunk 5 above)
Comment (chirag-yb) [reply docs/content/latest/releases/releases-overview.md:21]: @rkarthik007 do we want to advertise EOL & maintenance support dates for the latest release? I would imagine NO. 
I think adding the latest and future scheduled releases in the same table as supported stable releases are confusing. Can we keep the way it is for now and roll with it?

Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md]: The stable and latest releases releases are combined into one section and table.

File: docs/content/latest/releases/releases-overview.md (hunk 6 above)
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md]: Dates updated based on discussion with Chirag and comment by @rkarthik007

**Review by chirag-yb — APPROVED** [review]
looks good to me

**Comment by stevebang** [comment]
Moved content from this PR into #5711. Made it much easier to resolve odd merge conflicts.

File: docs/content/latest/releases/releases-overview.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+| :-------------------------------- | :----------- | :----------------------- | :----------- | :----------------------------------- |
+| [2.3](../whats-new)               | Sep 08, 2020 | ---                      | ---          | Latest (development and testing)
+| [2.2](../earlier-releases/v2.2.0) | Jul 15, 2020 | Jul 15, 2021             | Jan 15, 2022 | Stable (recommended for production)
+| [2.1](../earlier-releases/v2.1.0) | Feb 25, 2020 | Feb 25, 2021             | Aug 08, 2021 | Stable
+| [2.0](../earlier-releases/v2.0.0) | Sep 17, 2019 | Sep 17, 2020             | Mar 03, 2021 | Stable
+| [1.3](../earlier-releases/v1.3.0) | Jul 15, 2019 | Jul 15, 2020             | Jan 15, 2021 | Stable
+
+**Note:** The stable release series include `2.1`, `2.0`, and `1.3`, released prior to the new [release versioning](../) convention.
+
+## Planned releases
```
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:42]: @stevebang Minor changes on this section
1. Can we reword it to "Upcoming releases"
2. Can we flip the order of version and planned release columns and sort it by the release date?

File: docs/content/latest/releases/releases-overview.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+The following release series include:
+
+- [Stable](../versioning/#stable-releases): Supported for production deployments.
+- [Latest](../versioning/#latest-releases): Supported for development and testing only.
+
+[Release versioning](../versioning) gives details on the stable and latest release series numbering convention that started with `2.2`.
+
+[Planned releases](#planned-releases), in a section below, includes timelines for upcoming stable and latest release series.
+
+| Release series                    | Released     | Maintenance support ends | EOL          | Status                              |
```
Comment (chirag-yb) [inline docs/content/latest/releases/releases-overview.md:32]: @stevebang It would be nice to have the status (Latest (development and testing), Current Stable (recommended for production), Stable in parenthesis next to the release series instead of a separate column.

File: docs/content/latest/releases/versioning.md [hunk 9]
```diff
@@ -1,15 +1,41 @@
...
+
+## Release versioning convention
+
+Yugabyte follows the [semantic versioning (semver)](https://semver.org) convention for numbering release versions, modified slightly to use even and odd minor releases denote stable and development releases, respectively. Release versions follow the semantic versioning format of `MAJOR.MINOR.PATCH`, where non-negative integers are used for:
+
+- `MAJOR` — Includes substantial changes.
+- `MINOR` — Incremented when new features and changes are introduced.
+  - `EVEN` — Stable minor release, intended for production deployments.
+  - `ODD` — Latest minor release, intended for development and testing.
+- `PATCH` — bug fixes and revisions that do not break backwards compatibility.
```
Comment (chirag-yb) [inline docs/content/latest/releases/versioning.md:25]: @stevebang Lets remove this `that do not break backwards compatibility`

File: docs/content/latest/releases/releases-overview.md (hunk 7 above)
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:42]: done

File: docs/content/latest/releases/releases-overview.md (hunk 8 above)
Comment (stevebang) [reply docs/content/latest/releases/releases-overview.md:32]: done

File: docs/content/latest/releases/versioning.md (hunk 9 above)
Comment (stevebang) [reply docs/content/latest/releases/versioning.md:25]: Reworded before your comment, so take a look at what's there now. If needed, I can change it.
